### PR TITLE
Adding administrator registration functionality

### PR DESCRIPTION
**Pull request by Esteban-Naranjo**
It was modified the function "Create user" in order to used it in two cases, when is requested by a normal user and when is requested by an admin user.

### CODE REVIEW

File: cmd/api/main.go [hunk 1]
```diff
@@ -77,6 +77,10 @@ func main() {
 			user.POST("/", userHandler.CreateUser)
 			user.POST("/load-by-csv", userHandler.LoadUserByCSV)
 			user.POST("/reset-password", userHandler.ResetPassword)
+			admins := user.Group("/administrators", middleware.AdminMiddleware(authService))
+			{
+				admins.POST("/register-admin", userHandler.CreateUser)
+			}
```
Comment (SantiagoBedoya) [inline cmd/api/main.go]: We do not need to create a new route to create managers. It is a bad practice to use a handler for more than 1 route in the application. Please use the route on line 77 and delete these lines.

File: internal/adapter/middleware/admin_middleware.go
Comment (SantiagoBedoya) [inline internal/adapter/middleware/admin_middleware.go:1]: I understood what you did, but we don't need this middleware.

File: internal/adapter/middleware/auth_middleware.go [hunk 2]
```diff
@@ -31,5 +31,6 @@ func AuthMiddleware(authService port.AuthService) gin.HandlerFunc {
 
 		ctx.Set("userDNI", claims["sub"])
 		ctx.Set("userRole", claims["role"])
+		ctx.Set("adminPermission", false)
```
Comment (SantiagoBedoya) [inline internal/adapter/middleware/auth_middleware.go]: We do not need `AdminMiddleware` to know if the role is Admin or not. Please take a look at line 33, we have userRole, we can simply validate if this value has the value of Admin Role.

File: internal/core/service/auth_service.go
Comment (SantiagoBedoya) [inline internal/core/service/auth_service.go:1]: With the above comments, we do not need the `ValidateAdminRol` method.

**Review by SantiagoBedoya — CHANGES_REQUESTED** [review]
Please solve the comments.

**Review by SantiagoBedoyaU — APPROVED** [review]

**Pull request merged**